File: src/com/androhi/androiddrawableviewer/IconUtils.java
package com.androhi.androiddrawableviewer;

import javax.imageio.ImageIO;
import javax.swing.*;
import java.awt.*;
import java.io.File;
import java.io.IOException;

public class IconUtils {

    public static Icon createSmallIcon(String iconFile) {
        return createIcon(iconFile, 24);
    }

    public static Icon createOriginalIcon(String iconFile) {
        return createIcon(iconFile, 0);
    }

    private static Icon createIcon(String iconFile, int scaleSize) {
        File imageFile = new File(iconFile);
        try {
            Image originalImage = ImageIO.read(imageFile);
            if (scaleSize > 0) {
                Image resizedImage = originalImage.getScaledInstance(scaleSize, scaleSize, Image.SCALE_DEFAULT);
                return new ImageIcon(resizedImage);
            } else {
                return new ImageIcon(originalImage);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: src/main/kotlin/com/androhi/androiddrawableviewer/utils/IconUtils.kt
package com.androhi.androiddrawableviewer.utils

import java.awt.Image
import java.io.File
import java.io.IOException
import javax.imageio.ImageIO
import javax.swing.Icon
import javax.swing.ImageIcon

class IconUtils {
    companion object {
        fun createSmallIcon(iconFilePath: String): Icon? = createIcon(iconFilePath, 24)

        fun createOriginalIcon(iconFilePath: String): Icon? = createIcon(iconFilePath)

        private fun createIcon(iconFilePath: String, size: Int = 0): Icon? {
            val imageFile = File(iconFilePath)
            return try {
                val originalImage = ImageIO.read(imageFile)
                if (size > 0) {
                    val resizedImage = originalImage.getScaledInstance(size, size, Image.SCALE_DEFAULT)
                    ImageIcon(resizedImage)
                } else {
                    ImageIcon(originalImage)
                }
            } catch (e: IOException) {
                null
            }
        }
    }
}